qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -0,0 +1,327 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Quantum Kernel Algorithm"""
+
+from typing import Optional, Union
+
+import logging
+import numpy as np
+
+from qiskit import QuantumCircuit, QuantumRegister, ClassicalRegister
+from qiskit.circuit import ParameterVector
+from qiskit.circuit.library import ZZFeatureMap
+from qiskit.providers import Backend, BaseBackend
+from qiskit.utils import QuantumInstance
+from ..exceptions import QiskitMachineLearningError
+
+logger = logging.getLogger(__name__)
+
+
+class QuantumKernel:
+    r"""Quantum Kernel.
+
+    The general task of machine learning is to find and study patterns in data. For many
+    algorithms, the datapoints are better understood in a higher dimensional feature space,
+    through the use of a kernel function:
+
+    .. math::
+
+        K(x, y) = \langle f(x), f(y)\rangle.
+
+    Here K is the kernel function, x, y are n dimensional inputs. f is a map from n-dimension
+    to m-dimension space. :math:`\langle x, y \rangle` denotes the dot product.
+    Usually m is much larger than n.
+
+    The quantum kernel algorithm calculates a kernel matrix, given datapoints x and y and feature
+    map f, all of n dimension. This kernel matrix can then be used in classical machine learning
+    algorithms such as support vector classification, spectral clustering or ridge regression.
+    """
+
+    def __init__(self,
+                 feature_map: Optional[QuantumCircuit] = None,
+                 enforce_psd: bool = False,
```

Comment:
In the prior QSVM we had enforce_psd default as True so it would work by default with the qasm simulator. I would have thought perhaps True would be a better default here too. It may be worth stating here on the docstring that this is only done on when sampling using a qasm sim or real device i.e. when not using a statevector sim.